net/ipv4/tcp_output.c

```diff
@@ -2783,6 +2793,7 @@ static bool tcp_write_xmit(struct sock *sk, unsigned int mss_now, int nonagle,
 		if (unlikely(result)) {
 			if (result == -ENOMEM)
 				break; /* try again next time */
+			tcp_tfw_handle_error(sk, result);
```

Comment:
Why we don't handle `ENOMEM`, looks suspicious to continue working with connection if allocation error was catched?

Reply:
It was done previously. I think that we rely on the fact that if we catch ENOMEM here, we can try to call this function later, when we have enough memory